provider/github/watcher.go

```diff
@@ -73,34 +81,94 @@ func NewWatcher(transport http.RoundTripper, o *lookout.WatchOptions) (*Watcher,
 func (w *Watcher) Watch(ctx context.Context, cb lookout.EventHandler) error {
 	log.With(log.Fields{"urls": w.o.URLs}).Infof("Starting watcher")
 
-	repoIndex := 0
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	errCh := make(chan error)
+
+	for _, repo := range w.r {
+		go w.watchPrs(ctx, repo, cb, errCh)
+		go w.watchEvents(ctx, repo, cb, errCh)
+	}
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err := <-errCh:
+		return err
+	}
+}
 
+func (w *Watcher) watchPrs(ctx context.Context, repo *lookout.RepositoryInfo, cb lookout.EventHandler, errCh chan error) {
 	for {
-		repo := w.r[repoIndex]
-		repoIndex = (repoIndex + 1) % len(w.r)
+		resp, prs, err := w.doPRListRequest(ctx, repo.Username, repo.Name)
+		if err != nil && !NoErrNotModified.Is(err) {
+			errCh <- err
+			return
+		}
+
+		if err := w.handlePrs(cb, repo, resp, prs); err != nil {
+			if lookout.NoErrStopWatcher.Is(err) {
+				return
```

Comment:
NoErrStopWatcher comment says it should stop the main `.Watch` method, but with this return we are only stopping this goroutine while all the rest will continue. Right?

```
// NoErrStopWatcher if a new error of this kind is returned by EventHandler
// the Watcher.Watch function exits without error.
```

Reply:
Oh! Thank you! Very good catch!